nix/default.nix

```diff
@@ -3,12 +3,17 @@
 , config ? {}
 , overlays ? []
 , sourcesOverride ? {}
+, enableLibraryProfiling ? true
 }:
 let
   sources = import ./sources.nix { inherit pkgs; }
     // sourcesOverride;
   iohkNix = import sources.iohk-nix {};
-  haskellNix = import sources."haskell.nix" {};
+  haskellNix = import sources."haskell.nix" {
+    # Dev tools should be compiled with the same compiler as
+    # plutus increase sharing and keep closure sizes down
+    defaultCompilerNixName = "ghc883";
```

Comment:
I think this is unnecessary. Or if we do rely on it, I'd much prefer to pass the compiler name where it's needed rather than overriding the defaults.

Reply:
I did this to make sure `ghc865` is not used somewhere (even cabal-install for instance when compiled with ghc865 pulls in a different gcc and that makes the devcontainer larger).

Reply:
Hrm. Well, then at least we need to abstract this out somewhere so we're saying the compiler name once instead of many times as we do at the moment.

It's a bit annoying that we can't get it from `cabal.project` somehow.

Reply:
We could parse `cabal.project` and `cabal.project.local` contents in `cabalProject` and look for `compiler:`.  We are already pulling out the `source-repository-package` stuff.

Reply:
Hm, it seems like neither `compiler` nor `with-compiler` has the right semantics: `compiler` is "one of `ghc`, `ghcjs`...." i.e. not a version of ghc, and `with-compiler` is the actual path to a compiler binary... So it looks like there just isn't a correct way to specify that in `cabal.project` atm. Sad times.